### ISSUE TITLE
hoauthProcessUser callback hasn't the user profile as param

### ISSUE THREAD

**Issue by Darkheir** [SleepWalker/hoauth#44]
From what I see in the code the `hoauthProcessUser` hasn't the user profile in its params but if the user is new or not.
On the other side the `processUser` has the profile as a param.

``` php
if($this->alwaysCheckPass || $user->isNewRecord)
        if(method_exists($this->controller, 'hoauthProcessUser'))
            $user = $this->controller->hoauthProcessUser($user, $newUser);
    else
        $user = $this->processUser($user, $userProfile);
```

Wouldn't be better if `hoauthProcessUser` also had profile informations?

The main issue is that if we add a param to the callback it will break compatibility with older implemented callbacks.

**Comment by SleepWalker** [comment]
Yeah, that was not so clever from my side... for now it's better that you adjust this callback by yourself for your project.

In the next update, that I will make in the next month, I will change the signature of this callback and I will also add come new callbacks for the case, when user wants introduce very small changes without changing the entire user processing method.

P.S. Another option is to use UserOAuth to get the user's profile (cached or fresh)
